File: benchmarks/corpus/java/NearA.java
final class NearA {
    static long riskScore(long[] events) {
        long score = 7;
        for (int index = 0; index < events.length; index++) {
            long event = events[index];
            long weighted = event * 5;
            score += weighted;
        }
        score ^= events.length;
        return Math.max(0, score);
    }
}

File: benchmarks/corpus/rust/near_a.rs
fn risk_score(events: &[i64]) -> i64 {
    let mut score = 7_i64;
    for event in events {
        let weighted = event.saturating_mul(5);
        score = score.saturating_add(weighted);
        if score > 50_000 {
            score /= 2;
        }
    }
    score.clamp(-75_000, 75_000)
}
